backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -54,6 +55,37 @@ public function updateCard($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function buyCard($card_id)
+    {
+        //update card user's id with current id
+        $user = auth()->user();
+        $card = Card::find($card_id);
+        if ($card->isUserOwner($user)) {
+            return response()->build("User own the card");
+        }
+
+        $data = json_decode(Request::getContent(), true);
+        foreach ($data as $key => $value) {
```

Comment:
don't need to do this all in a loop if this will be the only value you're grabbing, 68:74 can just be short-ended to `$card->user_id = $data['user_id']`

Reply:
shouldn't this be $card->user_id = $user->user_id ?